### ISSUE TITLE
iOS8: no workie from homescreen shortcut

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#2547]
**[Mingle Card: 2822](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/2822)**
Steps to reproduce
------------------

```
| | |
|-|-|
|**Version #**|
| |
|**OS**|
|iOS8|
|**Browser**|
| |
|**Username**|
| |
|**Password**|
| |

  

 


```

1.  Launch FieldFX from homescreen shortcut (web clip)

Expected result
---------------

_Workie_

Actual Result
-------------

_No workie_

Analysis
--------

http://www.sencha.com/blog/apple-shows-love-for-html5-with-ios-8

https://github.com/axemclion/IndexedDBShim/issues/167

Prior to iOS8 there was UIWebView, the core browser engine that drives Mobile Safari and home screen apps. We recommended running FieldFX from a web clip because it got around the database size limitations.

iOS8 introduced a new engine called WKWebView that supports IndexedDB (Yay new hotness!). However, home screen apps still use the old UIWebView (Boo old-and-busted!). 

Unfortunately, IndexedDBShim no longer works because window.indexedDB now exists as a readonly property that can't be patched. In the case of WKWebView window.indexedDB works. In the case of UIWebView, window.indexedDB returns null.

So basically... we need a global getter that figures this out and use it everywhere we need window.indexedDB (including wrappers like indexedDoucheBag).

Related Cards

Test Plan
---------
